app/src/main/java/com/infomaniak/mail/ui/main/menu/SwitchMailboxesAdapter.kt

```diff
@@ -0,0 +1,73 @@
+/*
+ * Infomaniak kMail - Android
+ * Copyright (C) 2022-2023 Infomaniak Network SA
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package com.infomaniak.mail.ui.main.menu
+
+import android.view.LayoutInflater
+import android.view.ViewGroup
+import androidx.recyclerview.widget.RecyclerView
+import com.infomaniak.mail.MatomoMail.SWITCH_MAILBOX_NAME
+import com.infomaniak.mail.MatomoMail.trackAccountEvent
+import com.infomaniak.mail.MatomoMail.trackMenuDrawerEvent
+import com.infomaniak.mail.data.cache.RealmDatabase
+import com.infomaniak.mail.data.models.Mailbox
+import com.infomaniak.mail.databinding.ItemSwitchMailboxBinding
+import com.infomaniak.mail.ui.main.menu.SwitchMailboxesAdapter.SwitchMailboxesViewHolder
+import com.infomaniak.mail.utils.AccountUtils
+import com.infomaniak.mail.views.MenuDrawerItemView.SelectionStyle
+
+class SwitchMailboxesAdapter(
+    private var mailboxes: List<Mailbox> = emptyList(),
+    private val isInMenuDrawer: Boolean = true,
```

Comment:
To avoid mistakes, don't set a default value so the next dev has to think about what he really wants to do here

Reply:
I will change this in another PR, as it can be change in other places too

Reply:
https://github.com/Infomaniak/android-kMail/pull/849